eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/ReplicationEndpoint.scala

```diff
@@ -363,6 +359,96 @@ class ReplicationEndpoint(
+ */
+trait EndpointFilter {
+  def filterFor(targetLogId: String, sourceLogName: String): ReplicationFilter
+}
+
+object EndpointFilter {
+  private class CombiningEndpointFilter(
+    targetFilters: Map[String, ReplicationFilter],
+    sourceFilters: Map[String, ReplicationFilter],
+    targetSourceCombinator: (ReplicationFilter, ReplicationFilter) => ReplicationFilter) extends EndpointFilter {
+
+    override def filterFor(targetLogId: String, sourceLogName: String): ReplicationFilter = {
+      (targetFilters.get(targetLogId), sourceFilters.get(sourceLogName)) match {
+        case (None, Some(sourceFilter))               => sourceFilter
+        case (Some(targetFilter), None)               => targetFilter
+        case (None, None)                             => NoFilter
+        case (Some(targetFilter), Some(sourceFilter)) => targetSourceCombinator(targetFilter, sourceFilter)
+      }
+    }
+  }
+
+  /**
+   * Creates an [[EndpointFilter]] that computes a [[ReplicationFilter]] for a replication read request
+   * from a source log to a target log by and-combining target and source filters when given in the provided [[Map]]s.
+   * If only source or target filter is given that is returned. If no filter is given [[com.rbmhtechnology.eventuate.ReplicationFilter.NoFilter]] is returned.
+   *
+   * This [[EndpointFilter]] can be useful when an eventsourced service that is not replicated collaborates
+   * with different external service. In that case `sourceFilters` can ensure that no ''internal'' event is
+   * replicated to other services and `targetFilters` can add target specific filter criteria.
+   *
+   * @param targetFilters maps target log ids to the [[ReplicationFilter]] that shall be applied when replicating to the target log
+   * @param sourceFilters maps source log names to the [[ReplicationFilter]] that shall be applied when replicating from the source log
+   */
+  def targetAndSourceFilters(targetFilters: Map[String, ReplicationFilter], sourceFilters: Map[String, ReplicationFilter]): EndpointFilter =
+    new CombiningEndpointFilter(targetFilters, sourceFilters, _ and _)
+
+  /**
+   * Creates an [[EndpointFilter]] that computes a [[ReplicationFilter]] for a replication read request
+   * from a source log to a target log by returning the target filter when given in `targetFilters`. If only
+   * a source filter is given in `sourceFilters` that is returned otherwise [[com.rbmhtechnology.eventuate.ReplicationFilter.NoFilter]] is returned.
+   *
+   * This [[EndpointFilter]] can be useful when an eventsourced service that can be replicated collaborates
+   * with different internal services and has a fixed set of events for external communication.
+   * In that case `sourceFilters` can ensure that only external events are replicated to external services
+   * while in case of internal communication (with internal service or replication partners) these filters
+   * can be overwritten.
```

Comment:
I think most people will have difficulties to understand what you mean with *internal* and *external* here. This comes from an application domain but it is not a concept of Eventuate itself. I'd rather say that the use case is that some (more privileged) remote targets may replace a (default) source filter with a target-specific filter. In this context, I also find the term *service* a bit confusing. To me it would be clearer if the API docs mention *local source (log)* and *remote target (log)*.